data/1.0.1/dataset.json

```diff
@@ -6709,6 +6709,17 @@
       ],
       "uiSelectionGroup": "prospr"
     },
+    "coronal": {
+      "isExclusive": false,
+      "uiSelectionGroup": "anatomical-views",
+      "viewerTransform": {
```

Comment:
@constantinpape @K-Meech 
I am not sure this is the correct syntax. I think it should say `normalVector` where it says `viewerTransform`, shouldn't it?
Do we have a working example of a viewerTransform somewhere else?

Reply:
Java:

```
public class View
{
	private String uiSelectionGroup;
	private List< SourceDisplay > sourceDisplays;
	private List< SourceTransformer > sourceTransforms;
	private ViewerTransform viewerTransform;
	private boolean isExclusive = false;
	private String name;
```

...where ViewerTransform is an Interface, which gets resolved to an actual class by means of:

```public class ViewerTransformAdapter implements JsonSerializer<ViewerTransform>, JsonDeserializer< ViewerTransform >
{
	private static Map<String, Class> nameToClass = new TreeMap<>();
	private static Map<String, String> classToName = new TreeMap<>();

	static {
		nameToClass.put("affine", AffineViewerTransform.class);
		classToName.put(AffineViewerTransform.class.getName(), "affine");
		nameToClass.put("normalizedAffine", NormalizedAffineViewerTransform.class);
		classToName.put(NormalizedAffineViewerTransform.class.getName(), "normalizedAffine");
		nameToClass.put("position", PositionViewerTransform.class);
		classToName.put(PositionViewerTransform.class.getName(), "position");
		nameToClass.put("timepoint", TimepointViewerTransform.class);
		classToName.put(TimepointViewerTransform.class.getName(), "timepoint");
		nameToClass.put("normalVector", NormalVectorViewerTransform.class);
		classToName.put(NormalVectorViewerTransform.class.getName(), "normalVector");
	}
```

Reply:
It's the correct syntax: https://github.com/mobie/platybrowser-project/blob/main/data/1.0.1/dataset.json#L2854-L2872

Reply:
Oh, wow, now I get it. It looks like we derive the type of the class from the name of the `double[] parameter` field, which in this case is `normalVector`. Is this best practice?

Reply:
I can't speak to best practices in Java here, but I would like to avoid changing the spec fundementally for the viewerTransforms at this point.

Reply:
It's probably not best practice - but I'd agree with @constantinpape that it would be good to avoid changing it for now.